src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -759,24 +759,30 @@ private boolean isSameParameter(CtMethod<?> method, CtTypeReference<?> ctParamet
 				return false;
 			}
 		}
+		if (ctParameterType.equals(factory.Type().OBJECT) && (expectedType instanceof CtTypeParameterReference)) {
+			CtTypeParameterReference typeParameterReference = (CtTypeParameterReference) expectedType;
+			if (!typeParameterReference.isUpper() || (typeParameterReference.isUpper() && typeParameterReference.getBoundingType().equals(ctParameterType))) {
```

Comment:
in this `if` statement this is true `ctParameterType.equals(factory.Type().OBJECT)`, so call of `typeParameterReference.getBoundingType().equals(ctParameterType))` might be changed to
`typeParameterReference.getBoundingType().equals(factory.Type().OBJECT))` ... but may be it points to mistake on the line 762 ... and the condition `ctParameterType.equals(factory.Type().OBJECT)` should be changed?